makefiles/pseudomodules.inc.mk

```diff
@@ -298,6 +298,9 @@ PSEUDOMODULES += skald_eddystone
 # define optimized read function of DS18 driver as a pseudo module
 PSEUDOMODULES += ds18_optimized
 
+PSEUDOMODULES += crypto_aes_128
```

Comment:
Should `crypto_aes_128` be added as a `DEFAULT_MODULE` if `crypto` is used? I think that would match the current expected behaviour.

Reply:
Can you give an example of how this would work with `DEFAULT_MODULE` ? I can't find any documentation about it.

Why would this approach be better than my approach of setting `MODULE_CRYPTO_AES_128` if none of the pseudomodules has been specified ?

Reply:
> Can you give an example of how this would work with DEFAULT_MODULE ? I can't find any documentation about it.

https://github.com/RIOT-OS/RIOT/blob/569e299e63dc7dd132ff528b02b4c1a76d43f563/sys/Makefile.dep#L787-L788
What's added to `DEFAULT_MODULE` can be disabled by `DISABLE_MODULE`.

> Why would this approach be better than my approach of setting `MODULE_CRYPTO_AES_128` if none of the pseudomodules has been specified ?

Because just defining a macro on a specific header file is not the same as activating a module. The build system is in charge of resolving this dependencies, because it can affect other modules, configurations or even bring in other dependencies. The `MODULE_` macros are injected via `riotbuild.h`, so they are available at compilation to all modules, even if they don't include the ciphers header file.

Reply:
Okay I see. Thank you !

One more question though: If I add `crypto_aes_128` as `DEFAULT_MODULE` it will always be included. This will remove the optimization we added with `AES_KEY_SIZE(ctx)` if someone wants to use only 192 bit or only 256 bit keys.

Reply:
As it is added as `DEFAULT_MODULE` it can be disabled via `DISABLE_MODULE`, in case the user only needs AES 256. It's true that the optimization is lost if more than one module is included. But I think it's important to keep the current expected behaviour, meaning AES 128 is available when `crypto` is included.

Maybe some documentation on this would be useful, indicating the available modules and the default configuration.

Reply:
Updated the PR. I added documentation about the the behavior at the defines for the `AES_KEY_SIZE_` because this is where I would look for such information.